lib/rules/no-whitespace-before-property.js

```diff
@@ -25,14 +46,37 @@ module.exports = function(context) {
             var prop = node.property;
 
             if (astUtils.isTokenOnSameLine(obj, prop)) {
-                if (sourceCode.isSpaceBetweenTokens(obj, prop)) {
-                    context.report({
-                        node: node,
-                        message: "Unexpected whitespace before property '{{ propName }}'.",
-                        data: {
-                            propName: prop.name
+                var tokenBeforeProp;
+                var tokenBeforePunc;
+
+                tokenBeforeProp = sourceCode.getTokenBefore(prop);
+                while (tokenBeforeProp.value !== "." && tokenBeforeProp.value !== "[") {
+                    tokenBeforeProp = sourceCode.getTokenBefore(tokenBeforeProp);
+                }
+
+                tokenBeforePunc = sourceCode.getTokenBefore(tokenBeforeProp);
+
+                if (tokenBeforeProp.value === ".") {
```

Comment:
@kaicataldo `leftToken` and `rightToken` are different token in each case, so `if` statements to check the type of tokens are not needed :)

This should pass all tests: https://gist.github.com/mysticatea/8ad720c48dd9c67336e3